### PR TITLE
25543 docs cleanup

### PR DESCRIPTION
**Pull request by vipulnsward**
- Tone down documentation
- Remove repetative docs
- Fix grammar on sentences
- Add escaping for literals

### CODE REVIEW

File: actionview/lib/action_view/helpers/tag_helper.rb [hunk 1]
```diff
@@ -119,35 +119,35 @@ def method_missing(called, *args, &block)
       # ==== Passing content
       # Tags can pass content to embed within it:
       #
-      #   tag.h1 'All shit fit to print' # => <h1>All shit fit to print</h1>
+      #   tag.h1 'All titles fit to print' # => <h1>All titles fit to print</h1>
       #
       #   tag.div tag.p('Hello world!')  # => <div><p>Hello world!</p></div>
       #
-      # Content can also be captured with a block. Great for ERB templates:
+      # Content can also be captured with a block, which is useful in templates:
```
Comment (maclover7) [inline actionview/lib/action_view/helpers/tag_helper.rb]: in ERB templates

File: actionview/lib/action_view/helpers/tag_helper.rb [hunk 2]
```diff
@@ -119,35 +119,35 @@ def method_missing(called, *args, &block)
...
       #
       #   <%= tag.p do %>
       #     The next great American novel starts here.
       #   <% end %>
       #   # => <p>The next great American novel starts here.</p>
       #
       # ==== Options
-      # Any passed options becomes attributes on the generated tag.
+      # Any passed options become attributes on the generated tag.
```
Comment (maclover7) [inline actionview/lib/action_view/helpers/tag_helper.rb]: will become

File: actionview/lib/action_view/helpers/tag_helper.rb [hunk 3]
```diff
@@ -182,7 +182,7 @@ def method_missing(called, *args, &block)
       #   tag.br  # => <br>
       #
       # === Legacy syntax
-      # Following format is legacy syntax. It will be deprecated in future versions of rails.
+      # Following format is for legacy syntax support. It will be deprecated in future versions of rails.
```
Comment (maclover7) [inline actionview/lib/action_view/helpers/tag_helper.rb]: add a `the` at the start of the sentence, before `following`

`rails` --> Rails

**Comment by maclover7** [comment]
few comments, otherwise 🚢  from me :)

File: actionview/lib/action_view/helpers/tag_helper.rb (hunk 3 above)
Comment (kaspth) [inline actionview/lib/action_view/helpers/tag_helper.rb]: That ought to be `The following` 😁

File: actionview/lib/action_view/helpers/tag_helper.rb (hunk 2 above)
Comment (kaspth) [inline actionview/lib/action_view/helpers/tag_helper.rb]: @maclover7 Aren't docs meant to keep it to present tense?

File: actionview/lib/action_view/helpers/tag_helper.rb [hunk 4]
```diff
@@ -119,35 +119,35 @@ def method_missing(called, *args, &block)
       # ==== Passing content
       # Tags can pass content to embed within it:
       #
-      #   tag.h1 'All shit fit to print' # => <h1>All shit fit to print</h1>
+      #   tag.h1 'All titles fit to print' # => <h1>All titles fit to print</h1>
```
Comment (kaspth) [inline actionview/lib/action_view/helpers/tag_helper.rb]: You're killing me 😄

If we're changing this, why can't we go with 💩?

File: actionview/lib/action_view/helpers/tag_helper.rb [hunk 5]
```diff
@@ -202,15 +202,6 @@ def method_missing(called, *args, &block)
...
       #
-      # To play nicely with JavaScript conventions sub-attributes are dasherized.
-      # For example, a key +user_id+ would render as <tt>data-user-id</tt> and
-      # thus accessed as <tt>dataset.userId</tt>.
-      #
-      # Values are encoded to JSON, with the exception of strings, symbols and
-      # BigDecimals.
-      # This may come in handy when using jQuery's HTML5-aware <tt>.data()</tt>
-      # from 1.4.3.
-      #
```
Comment (kaspth) [inline actionview/lib/action_view/helpers/tag_helper.rb]: This is still applicable to the legacy syntax, no?

File: actionview/lib/action_view/helpers/tag_helper.rb (hunk 1 above)
Comment (vipulnsward) [inline actionview/lib/action_view/helpers/tag_helper.rb]: I left it out on purpose, it's not specific to ERB

File: actionview/lib/action_view/helpers/tag_helper.rb (hunk 5 above)
Comment (vipulnsward) [inline actionview/lib/action_view/helpers/tag_helper.rb]: @kaspth Yes, but these are descriptions of options, and about what options can be passed here. 
We already describe these options before, with exact same desciption.

**Comment by vipulnsward** [comment]
Update to take care of single comment, replied to the rest.

**Comment by vipulnsward** [comment]
Thanks for the review!

**Pull request merged**